tools/markdown_lint/linter.py

```
@@ -0,0 +1,816 @@
+        self.reports[file_path] = report
+
+        try:
+            # Read file content
+            content = file_path.read_text(encoding="utf-8")
+            lines = content.splitlines(keepends=False)
+
+            # Initialize state
+            in_code_block = False
+            in_html_comment = False
+            in_list = False
+            list_indent = 0
```

Ruff (0.12.2)

Local variable `list_indent` is assigned to but never used

Remove assignment to unused variable `list_indent`

(F841)